src/build.d

```diff
@@ -1037,6 +1032,15 @@ void parseEnvironment()
             default:
                 throw abortBuild(format("Variable 'PIC' should be '0', '1' or <empty> but got '%s'", picValue));
         }
+        version (X86)
+        {
+            // https://issues.dlang.org/show_bug.cgi?id=20466
+            static if (__VERSION__ < 2090)
+            {
+                pragma(msg, "Warning: PIC build disabled because of Issue 20466!");
```

Comment:
```suggestion
                pragma(msg, "Warning: PIC will be off by default for this build of DMD because of Issue 20466!");
```

Would be more accurate. But even this feels wrong, because `HOST_DMD` is not guaranteed to be what builds `src/build.d`...

Reply:
That would be a hypothetical `BUILD_DMD`? Or building dmd in a way that isn't supported by either our make files or build scripts?

Reply:
> That would be a hypothetical `BUILD_DMD`?

If you mean "the compiler that is used to build `src/build.d`", then yes. `BUILD_DMD` could be the latest but `HOST_DMD` an older version (e.g. oldest bootstrap), in which case the detection wouldn't work.

Reply:
No, the other way around.  I.e:  `HOST_DMD` builds all programs to that run on the host (`build.d`, `run.d`, `d_do_test.d`, etc...), `BUILD_DMD` is the compiler that builds dmd.

Reply:
Because of PIC being disabled for older compilers, there is every chance that pipelines will fail if I switch the image to 20.04 too.  Let's deal with that when we reach that hurdle though. :-)

In the worse case, the dmd baseline has to be raised to 2.090. :-(

Reply:
Or even worse 2.097.1, because that's when the libraries started to be built with `-fPIC`.